### DISCUSSION TITLE
Do we even need `func` keyword?

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4700] [Scripting]
```swift
var num = 1

do_something1():
    pass

static do_something2():
    pass

_ready():
    do_something1()
```

Looks kinda unusual, but imo it's clear that `do_something1` and `do_something2` are functions.

**Comment by dalexeev** [comment]
Personally, I find this less obvious and less readable since declarations keywords are usually highlighted in a bright color or bold text. The `var name: Type = value` syntax is used in many languages. The same goes for function declarations, although the keyword may be different (`function`, `func`, `fun`, `fn`, `procedure`, `proc`, `def`). C and C++ are exceptions, but they favor conciseness over friendliness.

We prefer not to have multiple syntax options _doing the same thing_. For example, we have two dictionary styles, but the Lua style allows you to omit quotes for keys. We have two setter/getter styles, but the alternative style allows you to use a separate function like `setget` in 3.x. An alternative function declaration syntax would not bring any bonuses other than aesthetic ones. I think it's not worth it, it would only complicate parser, clutter documentation, etc.

Also, judging by the reactions and activity over almost 2 years, this proposal did not receive support. So I think it's reasonable to close this discussion. Thanks for the contribution nonetheless!
